improver_tests/wxcode/wxcode/test_WeatherSymbols.py

```diff
@@ -921,13 +936,17 @@ def test_weather_data_global(self):
         data_cld_low = np.zeros((1, 3, 3), dtype=np.float32)
         data_vis = np.zeros((2, 3, 3), dtype=np.float32)
         data_precip = np.max(np.array([data_snow, data_rain]), axis=0)
+        data_convective_ratio = np.array(
+            [[[1.0, 0.0, 1.0], [0.0, 1.0, 0.0], [1.0, 0.0, 0.0]],], dtype=np.float32,
+        )
```

Comment:
This may seem pedantic but it's important.  I've had a look at this file more widely, in conjunction with the weather symbols decision tree documentation, and I don't think it ever tests the convective or heavy precipitation tree.  This is because the "cloud" input data fields are zero in both places where the convective ratio probability is high, and where most of the precipitation probabilities are highest.  As I read the decision tree, this means the precipitation will always follow the "drizzle" path because the P(Cloudy) < 0.5.

Having looked at the file, it does look like a lot of paths are not covered which we might want.  I'm wondering whether a bit of follow on work might be worth doing, to review the coverage of these tests in terms of output symbols and make sure the inputs are consistent?  @bayliffe ?